### ISSUE TITLE
Meteor update error. Should fail with better output on server issues

### ISSUE THREAD

**Issue by fungilation** [meteor/meteor#4921]
```
Octopus:~/Documents/code/app2$ meteor update
This project is already at Meteor 1.1.0.3, the latest release.
                                              
~/.meteor/packages/meteor-tool/.1.1.4.o64jkk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/fibers/future.js:245
						throw(ex);
						      ^
Error: Must throw non-empty error
    at Object.Future.throw (~/.meteor/packages/meteor-tool/.1.1.4.o64jkk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/fibers/future.js:259:10)
    at ~/.meteor/packages/meteor-tool/.1.1.4.o64jkk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:541:23
    at runWithEnvironment (~/.meteor/packages/meteor-tool/.1.1.4.o64jkk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:139:21)
    at ~/.meteor/packages/meteor-tool/.1.1.4.o64jkk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:146:14
    at ~/.meteor/packages/meteor-tool/.1.1.4.o64jkk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:549:11
```

It just happened without any code modification since today. Any idea on what's wrong? Doesn't look like the fault of update to any particular non-core Meteor packages.

**Comment by devonbarrett** [comment]
My hunch is that it is related to S3 having problems. Quite a few people have reported the same issue.

**Comment by fungilation** [comment]
Hmm, error went away with updating my other app (app1) before going back to meteor update the above app in question (app2). FYI:

```
Octopus:~/Documents/code/app2$ cd ../app1/
Octopus:~/Documents/code/app1$ meteor update
                                              
Changes to your project's package version selections from updating the release:
                                              
facebook  upgraded from 1.2.0 to 1.2.1        

app1: updated to Meteor 1.1.0.3.          
                                              
Changes to your project's package version selections from updating package versions:
                                              
aldeed:autoform          upgraded from 5.3.2 to 5.4.0
fortawesome:fontawesome  upgraded from 4.3.0 to 4.4.0
mdg:camera               upgraded from 1.1.4 to 1.1.5
meteorhacks:fast-render  upgraded from 2.7.1 to 2.8.1
meteorhacks:inject-data  upgraded from 1.3.0 to 1.4.0
meteorhacks:kadira       upgraded from 2.22.0 to 2.23.0
meteortoys:toykit        upgraded from 0.8.2 to 0.8.7
momentjs:moment          upgraded from 2.10.3 to 2.10.6
msavin:mongol            upgraded from 1.1.0 to 1.1.5
tmeasday:publish-counts  upgraded from 0.4.0 to 0.6.0

Octopus:~/Documents/code/app1$ cd ../app2/
Octopus:~/Documents/code/app2$ meteor update
This project is already at Meteor 1.1.0.3, the latest release.
                                              
Changes to your project's package version selections from updating package versions:
                                              
fortawesome:fontawesome  upgraded from 4.3.0 to 4.4.0
meteorhacks:fast-render  upgraded from 2.7.1 to 2.8.1
meteorhacks:inject-data  upgraded from 1.3.0 to 1.4.0
meteortoys:toykit        upgraded from 0.8.6 to 0.8.7
tmeasday:publish-counts  upgraded from 0.5.1 to 0.6.0
```

**Comment by fungilation** [comment]
Or it was just temporary problem with S3

**Comment by naps62** [comment]
It really looks like an S3 issue: https://downdetector.com/status/aws-amazon-web-services

It happened to work for me for a couple of minutes, but then it started failing again

**Comment by devonbarrett** [comment]
@fungilation Probably worth keeping this open. If it is related to S3 then there needs to be better exception handling in place anyway.

**Comment by naps62** [comment]
@fungilation Are you sure it's working for you now? I still get the error on all my apps (and I did try to `meteor update`)

**Comment by fungilation** [comment]
Yup. No more error for me running meteor update

**Comment by naps62** [comment]
For me it's just working on one of my two apps, and even then, it only works on the master branch

As soon as I move to the branch I'm working on (which has a new package), I can't run the server or `meteor update`, but with an different error:

```
 ± meteor update
This project is already at Meteor 1.1.0.3, the latest release.
                                              
/home/naps62/.meteor/packages/meteor-tool/.1.1.4.xvre8c++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/fibers/future.js:245
                                                throw(ex);
                                                      ^
```

That's it, no stack trace. Nothing

**Comment by tmds** [comment]
I had the same issue today. The root cause of the empty error is this line: https://github.com/meteor/meteor/blob/devel/tools/utils/http-helpers.js#L333 because the server returns 503 (Service Unavailable) with an empty body.

**Comment by sebakerckhof** [comment]
Explanation: https://news.ycombinator.com/item?id=10033187
But indeed, this error should be handled better by meteor tool

**Comment by glasser** [comment]
Fixed (see #4922).
